addons/MountRoulette/MountRoulette.lua

```diff
@@ -0,0 +1,79 @@
+_addon.name    = 'Mount Roulette'
+_addon.author  = 'Dean James (Xurion of Bismarck)'
+_addon.version = '3.0.0'
+_addon.commands = {'mountroulette', 'mr'}
+
+require('lists')
+resources = require('resources')
+
+math.randomseed(os.time())
+
+allowed_mounts = L{}
+possible_mounts = L{}
+for _, mount in ipairs(resources.mounts) do
+    possible_mounts:append(mount.name:lower())
+end
+
+function update_allowed_mounts()
+    local kis = windower.ffxi.get_key_items()
+
+    for _, id in ipairs(kis) do
+        local ki = resources.key_items[id]
+        if ki.category == 'Mounts' and ki.name ~= "trainer's whistle" then -- Don't care about the quest KI
+            local mount_index = possible_mounts:find(function(possible_mount)
+                return windower.wc_match(ki.name:lower(), '♪' .. possible_mount .. '*')
+            end)
+            local mount = possible_mounts[mount_index]
+
+            -- Add this to allowed mounts if it is not already there
+            if not allowed_mounts:contains(mount) then
+                allowed_mounts:append(mount)
+            end
+        end
+    end
+end
+
+windower.register_event('load', function()
+    update_allowed_mounts()
+end)
+
+windower.register_event('incoming chunk', function(id)
+    if id == 0x055 then --ki update
+        update_allowed_mounts()
```

Comment:
Have you verified that this works once you get a new mount? I don't think it should, since the received KI packet will trigger the function, but at that point it hasn't reached the game yet, so `windower.ffxi.get_key_items()` would not know about it, I think. You may want to run it in a slightly delayed coroutine:
```lua
update_allowed_mounts:schedule(0.2)
```

Reply:
Ahh I did not know that - I've probably not ran into this issue due to getting new mounts in Upper Jeuno, then needing to go outside to summon a mount. Given this, do still think it's worth the schedule call?

Reply:
Thinking about it, EmpyPopTracker has this same pattern and works ok. Is it more of a race condition?

Reply:
Actually, now that I think about it this should work after all. We store our own copy of the key items, and `windower.ffxi.get_key_items` returns that, not the game's key item store, so it shouldn't matter. At least I think so...

If you have some mounts left to get you could test it by the function print out the mounts it has found, then you'll see if it's really there.

Reply:
I think an alt of mine has some mounts to turn in so I'll check that soon.

Reply:
Thankfully it seems to work fine:

![image](https://user-images.githubusercontent.com/606627/89741336-52560d00-da88-11ea-989c-321506a6a068.png)